<?xml version="1.0" ?>
<launch>
  <arg name="config"		default="aist"/>
  <arg name="scene"		default="calibration"/>
  <arg name="sim"		default="false"/>
  <arg name="robot_name"	default="b_bot"/>
  <arg name="camera_name"	default="a_phoxi_m_camera"/>
  <arg name="use_dual_quaternion" default="false"/>

  <!-- Start controllers for robots, cameras and other sensors -->
  <include file="$(find aist_bringup)/launch/$(arg config)_bringup.launch">
    <arg name="scene"		value="$(arg scene)"/>
    <arg name="sim"		value="$(arg sim)"/>
    <arg name="rviz_config"	value="$(find aist_handeye_calibration
				       )/launch/handeye_calibration.rviz"/>
  </include>

  <!-- Start ArUco -->
  <arg name="marker_id"		value="26"	unless="$(arg eye_on_hand)"/>
  <arg name="marker_id"		value="32"	if    ="$(arg eye_on_hand)"/>
  <arg name="marker_size"	value="0.07"	unless="$(arg eye_on_hand)"/>
  <arg name="marker_size"	value="0.16"	if    ="$(arg eye_on_hand)"/>
  <arg name="camera_frame"	value="dummy_camera"/>
  <arg name="marker_frame"	value="dummy_marker"/>
  <include file="$(find aist_routines)/launch/aruco_detector.launch">
    <arg name="marker_id"	value="$(arg marker_id)"/>
    <arg name="marker_size"	value="$(arg marker_size)"/>
    <arg name="camera_name"	value="$(arg camera_name)"/>
    <arg name="camera_frame"	value="$(arg camera_frame)"/>
    <arg name="marker_frame"	value="$(arg marker_frame)"/>
  </include>

  <!-- Dummy calibration to have a fully connected TF tree and
       see all frames -->
  <arg name="eye_on_hand"
       value="$(eval camera_name.find(robot_name) == 0)"/>
  <arg name="robot_base_frame"		value="workspace_center"/>
  <arg name="robot_effector_frame"	value="$(arg robot_name)_ee_link"/>
  <node name="dummy_handeye" pkg="tf" type="static_transform_publisher"
        args="0 0 0.05 0 0 0 $(arg robot_effector_frame) $(arg camera_frame) 10"
	if="$(arg eye_on_hand)"/>
  <node name="dummy_handeye" pkg="tf" type="static_transform_publisher"
        args="-0.425 0.45 0.88 -1.5707 0 -2.8 $(arg robot_base_frame) $(arg camera_frame) 10"
	unless="$(arg eye_on_hand)"/>

  <!-- Start easy_handeye calibrator -->
  <group ns="$(arg camera_name)_from_$(arg robot_name)">
    <param name="eye_on_hand"		value="$(arg eye_on_hand)"/>
    <param name="robot_base_frame"	value="$(arg robot_base_frame)"/>
    <param name="robot_effector_frame"	value="$(arg robot_effector_frame)"/>
    <param name="tracking_base_frame"	value="$(arg camera_frame)"/>
    <param name="tracking_marker_frame"	value="$(arg marker_frame)"/>
    <param name="use_dual_quaternion"	value="$(arg use_dual_quaternion)"/>

    <node name="handeye_calibrator"
	  pkg="aist_handeye_calibration" type="calibrator" output="screen"/>
  </group>

</launch>
